### DISCUSSION TITLE
[Qt client] Pressing TAB to auto-complete remote paths in thin-client mode

### DISCUSSION THREAD

**Discussion by zhen-huan-hu** [transmission/transmission#4789] [Ideas]
In Deluge thin-client mode, pressing TAB will auto-complete the partially-entered paths, like in Bash. Is this something that can be implemented in Transmission Qt client as well?

[Desktop 2023.02.09 - 20.18.33.01-1-1.webm](https://user-images.githubusercontent.com/9272721/217985171-16d8bcb5-98f8-42f8-b6d0-10da4bd89cec.webm)

**Comment by Ercsion** [comment]
Take the [RelocateDialog.cc](https://github.com/transmission/transmission/blob/main/qt/RelocateDialog.cc#L63) as an example, it contains an input edit `newLocationEdit` to get the new path for the remote session, which in the local session will turn into a button `newLocationButton` for the user to select a folder.
Mainly talk about remote sessions (thin client mode) here.
In my opinion, an RPC interface needs to be added in [RPC specification](https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md) to query the server for files and paths such as `path-check`.But how to limit the scope of the query? Allowing to query the full path of the server will increase the risks.
@ckerr

**Reply by zhen-huan-hu** [reply]
Shouldn't this be determined by the ownership of the process?
